tests/core/fsck-on-boot/task.yaml

```diff
@@ -0,0 +1,86 @@
+summary: the boot base provides essential fsck programs
+# TODO:UC20: this test fails on core16, core18 and core20 and may require
+# separate fixes on each platform. really fails on core20, fsck is not invoked!
+details: |
+  Snapd uses vfat on certain essential boot partitions, due to external
+  requirements imposed by the bootloader architecture. This test verifies that
+  the boot process is capable of detecting unclean fvat and fixing it before
+  such file system is mounted. This is an essential property to ensure
+  longevity of devices that rely on write to vfat to operate.
+prepare: |
+  tests.cleanup prepare
+execute: |
+  unmount_vfat() {
+    case "$SPREAD_SYSTEM" in
+      ubuntu-core-16-*)
+        # Refer to the core 16 PC gadget for details:
+        # https://github.com/snapcore/pc-amd64-gadget/blob/16/gadget.yaml
+        umount /boot/efi
+        ;;
+      ubuntu-core-18-*)
+        # Refer to the core 18 PC gadget for details:
+        # https://github.com/snapcore/pc-amd64-gadget/blob/18/gadget.yaml
+        umount /boot/efi
+        umount /boot/grub
+        ;;
+      ubuntu-core-20-*)
+        # TODO:UC20 The property of having to keep a mounted vfat at all time
+        # is not the most fortunate. Any power loss will result in a dirty
+        # filesystem. Could ubuntu-seed be re-mounted read-only at some point
```

Comment:
we have other TODO's about doing this, the issue is that some operations require ubuntu-seed to be writable, but I think the thinking there was that we would just re-mount as writable to do the things we need, then re-mount as read-only

Reply:
Yeah, I think this is preferred. Ideally we'd mount the partition as read only to begin with. Then re-mount as writable only inside snapd, around specific operations. This should decrease the chance of corruption due to accidental power-cycle significantly.